server/runtime/src/main/java/org/infinispan/server/security/ElytronSubjectSaslServer.java

```diff
@@ -0,0 +1,46 @@
+package org.infinispan.server.security;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.sasl.SaslServer;
+
+import org.infinispan.server.core.security.AuthorizingCallbackHandler;
+import org.infinispan.server.core.security.SubjectSaslServer;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.sasl.WildFlySasl;
+
+/**
+ * A {@link SaslServer} which, when complete, can return a negotiated property named {@link #SUBJECT} which contains a
+ * populated {@link Subject} representing the authenticated user.
+ *
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 10.0
+ **/
+public class ElytronSubjectSaslServer extends SubjectSaslServer {
```

Comment:
Package-private class?

Also, I noticed that the server doesn't really make use of `.impl` packages. Are we taking the approach that none of the server code is supported for direct use externally? Or is this something we need to lock down more.

Reply:
Indeed, the server code is not for direct use